### ISSUE TITLE
Getting errors on startup with v1.22-rc.1

### ISSUE THREAD

**Issue by njkevlani** [fatih/vim-go#2658]
### What did you do? (required: The issue will be **closed** when not provided)
Installed vim-go v1.22-rc.1 https://github.com/fatih/vim-go/releases/tag/v1.22-rc.1

### What did you expect to happen?
No Error should happen while opening a `.go` file
### What happened instead?
Getting Errors:

```
Error detected while processing /home/nilesh/.vim/plugged/vim-go/ftplugin/go.vim:
line   79:
E970: Unknown highlight group name: 'goSameId'
vim-go: initializing gopls
Error detected while processing function go#lsp#ClearDiagnosticHighlights[1]..go#util#ClearHighlights:
line    2:
E971: Property type goDiagnosticError does not exist
vim-go: initialized gopls
vim-go: function 4[10]..7[3]..8[2]..9[46]..<SNR>46_highlightMatches[7]..go#util#ClearHighlights, line 2: Vim(return):E971: Property type goDiagnosticError does not exist
```

```
Error detected while processing BufWinEnter Autocommands for "<buffer=2>":
E121: Undefined variable: b:go_diagnostic_matches
E116: Invalid arguments for function <SNR>46_highlightMatches  
```
#### vim-go version:
v1.22-rc.1

####  `vimrc` you used to reproduce (use a *minimal* vimrc with other plugins disabled; do not link to a 2,000 line vimrc):
<details><summary>vimrc</summary><br><pre>

call plug#begin()
Plug 'fatih/vim-go', { 'tag': 'v1.21' }
Plug 'tomasr/molokai'
call plug#end()

" Delete trailing white space on save
fun! CleanExtraSpaces()
  let save_cursor = getpos(".")
  let old_query = getreg('/')
  silent! %s/\s\+$//e
  call setpos('.', save_cursor)
  call setreg('/', old_query)
endfun

autocmd BufWritePre * :call CleanExtraSpaces()

" Open where I left it.
autocmd BufReadPost * if line("'\"") > 0 && line("'\"") <= line("$") && &ft !~# 'commit' | exe "normal! g'\"" | endif

colorscheme molokai
syntax enable
filetype plugin on
filetype indent on
set nocompatible
set cursorline
set number
set relativenumber
set autoread
set scrolloff=7
set ignorecase
set smartcase
set hlsearch
set lazyredraw
set showmatch
set mat=2
set t_co=256
set background=dark
set encoding=utf-8
set ffs=unix,dos,mac
set nobackup
set nowb
set noswapfile
set expandtab
set smarttab
set shiftwidth=2
set tabstop=2
set lbr
set tw=500
set ai "Auto indent
set si "Smart indent
set wrap "Wrap lines
set clipboard=unnamedplus
set titlestring=%t%(\ %r%M%w%)%(\ (%{expand(\"%:p:h\")})%)%(\ %a%)
set title
set ruler
set rulerformat=%9(%l/%L\ %c%)
set wildmenu
set wildmode=longest:full,full
set backspace=eol,start,indent
set hidden
set whichwrap+=<,>,[,]
set incsearch
set magic
set undofile
set undodir=~/.vim/undo
set list
set listchars=tab:\┆│\ ,
set showcmd
set completeopt=longest,menuone
set noshowmode
set belloff=esc,showmatch,wildmode

let g:go_highlight_types = 1
let g:go_highlight_operators = 1
let g:go_highlight_functions = 1
let g:go_highlight_function_calls = 1
let g:go_highlight_fields = 1
let g:go_highlight_extra_types = 1
let g:go_highlight_build_constraints = 1
let g:go_highlight_generate_tags = 1

let g:Illuminate_delay = 1000

nnoremap n nzzzv
nnoremap N Nzzzv
cmap w!! w !sudo tee > /dev/null %
nmap <Home> ^
vmap <Home> ^
imap <Home> <ESC>^i

</pre></details>

#### Vim version (first three lines from `:version`):

VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Jan  7 2020 22:53:41)
#### Go version (`go version`):

go version go1.13.6 linux/amd64

#### Go environment
<details><summary><code>go env</code> Output:</summary><br><pre>

GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/nilesh/.cache/go-build"
GOENV="/home/nilesh/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/nilesh/code/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/lib/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build704570322=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

#### gopls version
<details><summary><code>gopls version</code> Output:</summary><br><pre>

golang.org/x/tools/gopls 0.2.0
    golang.org/x/tools/gopls@v0.2.0 h1:ddCHfScTYOG6auAcEKXCFN5iSeKSAnYcPv+7zVJBd+U=

</pre></details>

**Comment by bhcleek** [comment]
How are you opening a go file?

Which version of Vim are you using? I see that you reported 8.2, but _which_ version of 8.2? The first 3 lines of the output of `:version` should provide the information I need.

**Comment by njkevlani** [comment]
> How are you opening a go file?

```:e file.go```

> Which version of Vim are you using?

```
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Jan  7 2020 22:53:41)
Included patches: 1-100
Compiled by Arch Linux
```

**Comment by bhcleek** [comment]
What does `:set rtp?` report?

Can you confirm that your `/home/nilesh/.vim/plugged/vim-go/syntax/go.vim` defines `goSameId`?

**Comment by njkevlani** [comment]
> What does `:set rtp?` report?

```
  runtimepath=~/.vim,~/.vim/plugged/nerdtree/,~/.vim/plugged/vim-commentary/,~/.vim/plugged/ctrlp.vim/,~/.vim/plugged/vim-go/,~/.vim/plugged/vim-fugitive/,~/.vim/plugged/ultisnips/,~/.vim/pl
ugged/molokai/,~/.vim/plugged/vim-illuminate/,~/.vim/plugged/delimitMate/,~/.vim/plugged/vim-repeat/,~/.vim/plugged/vim-surround/,~/.vim/plugged/vim-signify/,/usr/share/vim/vimfiles,/usr/sha
re/vim/vim82,/usr/share/vim/vimfiles/after,~/.vim/plugged/ultisnips/after,~/.vim/after
```

> Can you confirm that your `/home/nilesh/.vim/plugged/vim-go/syntax/go.vim` defines `goSameId`?

My `go.vim` is same as [go.vim at v1.22-rc.1](https://github.com/fatih/vim-go/blob/v1.22-rc.1/syntax/go.vim). Do I have to look somewhere else?

**Comment by bhcleek** [comment]
> My go.vim is same as go.vim at v1.22-rc.1. Do I have to look somewhere else?

No, that should be enough. Notice that `goSameId` is set in that file: https://github.com/fatih/vim-go/blob/v1.22-rc.1/syntax/go.vim#L392

I wonder if you have a syntax file earlier in your runtimepath than vim-go that's interfering somehow. 🤔 Can you try duplicating this with out loading all the other plugins? Strangely, the vimrc you posted only load vim-go and molokai and doesn't set runtimepath, yet your runtimepath clearly has much more loaded.

**Comment by cnbuff410** [comment]
I'm having the same issue, and my message is

> Error detected while processing /Users/kunli/.vim/bundle/vim-go/ftplugin/go.vim:
> line   79:
> E970: Unknown highlight group name: 'goSameId'

Seems like my vim-go is trying to search the goSameId from different file?

**Comment by njkevlani** [comment]
> Strangely, the vimrc you posted only load vim-go and molokai and doesn't set runtimepath, yet your runtimepath clearly has much more loaded.

Yes, because I posted `.vimrc` without other plugins. But I posted output of `set rtp` with my regular `.vimrc`. My bad.

Output of `:set rtp` with posted `.vimrc` is

```
  runtimepath=~/.vim,~/.vim/plugged/vim-go/,~/.vim/plugged/molokai/,/usr/share/vim/vimfiles,/usr/share/vim/vim82,/usr/share/vim/vimfiles/after,~/.vim/after
```

The problem persists with regular this posted `.vimrc` as well.

**Comment by lukekhamilton** [comment]
same issue here. Fresh install osx and vim. 

```
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Jan  8 2020 00:57:48)
macOS version
Included patches: 1-100
Compiled by Homebrew
```

```
  runtimepath=~/.vim,~/.vim/bundle/Vundle.vim,~/.vim/bundle/vim-go,/usr/local/share/vim/vimfiles,/usr/local/share/vim/
vim82,/usr/local/share/vim/vimfiles/after,~/.vim/after,~/.vim/bundle/Vundle.vim,~/.vim/bundle/Vundle.vim/after,~/.vim/
bundle/vim-go/after
```

```
set nocompatible              " be iMproved, required
filetype off                  " required

set rtp+=~/.vim/bundle/Vundle.vim
call vundle#begin()

Plugin 'VundleVim/Vundle.vim'
Plugin 'fatih/vim-go'

call vundle#end()            " required
filetype plugin indent on    " required
syntax on 
set clipboard=unnamed
set path+=**
set wildmenu
set wildmode=longest:full,full
set hlsearch
set incsearch
```

**Comment by bhcleek** [comment]
I'm using vim 8.2.100, too, and I'm not seeing this error despite trying multiple replication paths.

Can you run `:scriptnames` after opening a .go file and report its full output here?

**Comment by njkevlani** [comment]
> Can you run `:scriptnames` after opening a .go file and report its full output here?

<details><summary>scriptnames</summary><br><pre>
  1: /etc/vimrc
  2: /usr/share/vim/vimfiles/archlinux.vim
  3: ~/.vimrc
  4: /usr/share/vim/vimfiles/autoload/plug.vim
  5: /usr/share/vim/vim82/filetype.vim
  6: ~/.vim/plugged/vim-go/ftdetect/gofiletype.vim
  7: /usr/share/vim/vimfiles/ftdetect/PKGBUILD.vim
  8: /usr/share/vim/vimfiles/ftdetect/dockerfile.vim
  9: /usr/share/vim/vim82/ftplugin.vim
 10: /usr/share/vim/vim82/indent.vim
 11: /usr/share/vim/vim82/syntax/syntax.vim
 12: /usr/share/vim/vim82/syntax/synload.vim
 13: /usr/share/vim/vim82/syntax/syncolor.vim
 14: ~/.vim/plugged/molokai/colors/molokai.vim
 15: /usr/share/vim/vim82/syntax/nosyntax.vim
 16: ~/.vim/plugged/vim-go/plugin/go.vim
 17: ~/.vim/plugged/vim-go/autoload/go/config.vim
 18: ~/.vim/plugged/vim-go/autoload/go/util.vim
 19: /usr/share/vim/vimfiles/plugin/fzf.vim
 20: /usr/share/vim/vim82/plugin/getscriptPlugin.vim
 21: /usr/share/vim/vim82/plugin/gzip.vim
 22: /usr/share/vim/vim82/plugin/logiPat.vim
 23: /usr/share/vim/vim82/plugin/manpager.vim
 24: /usr/share/vim/vim82/plugin/matchparen.vim
 25: /usr/share/vim/vim82/plugin/netrwPlugin.vim
 26: /usr/share/vim/vim82/plugin/rrhelper.vim
 27: /usr/share/vim/vim82/plugin/spellfile.vim
 28: /usr/share/vim/vim82/plugin/tarPlugin.vim
 29: /usr/share/vim/vim82/plugin/tohtml.vim
 30: /usr/share/vim/vim82/plugin/vimballPlugin.vim
 31: /usr/share/vim/vim82/plugin/zipPlugin.vim
 32: ~/.vim/plugged/vim-go/ftplugin/go.vim
 33: ~/.vim/plugged/vim-go/compiler/go.vim
 34: /usr/share/vim/vim82/compiler/go.vim
 35: ~/.vim/plugged/vim-go/ftplugin/go/commands.vim
 36: ~/.vim/plugged/vim-go/ftplugin/go/mappings.vim
 37: ~/.vim/plugged/vim-go/ftplugin/go/snippets.vim
 38: ~/.vim/plugged/vim-go/ftplugin/go/tagbar.vim
 39: /usr/share/vim/vim82/ftplugin/go.vim
 40: ~/.vim/plugged/vim-go/indent/go.vim
 41: /usr/share/vim/vim82/indent/go.vim
 42: ~/.vim/plugged/vim-go/syntax/go.vim
 43: /usr/share/vim/vim82/syntax/go.vim
 44: ~/.vim/plugged/vim-go/autoload/go/lsp.vim
 45: ~/.vim/plugged/vim-go/autoload/go/auto.vim
 46: ~/.vim/plugged/vim-go/autoload/go/template.vim
 47: ~/.vim/plugged/vim-go/autoload/go/tool.vim
 48: ~/.vim/plugged/vim-go/autoload/go/path.vim
 49: ~/.vim/plugged/vim-go/autoload/go/guru.vim
</pre></details>

**Comment by skipcloud** [comment]
I'm having a similar problem with v1.22. These are the errors I see with my normal vimrc
```
Error detected while processing function go#guru#ClearSameIds[1]..go#util#ClearHighlights:
line    2:
E119: Not enough arguments for function: prop_remove
Error detected while processing function go#lsp#ClearDiagnosticHighlights[1]..go#util#ClearHighlights:
line    2:
"indexing/index_menu_items_for_zone/main.go" [noeol] 11L, 198C
Error detected while processing function go#guru#ClearSameIds[1]..go#util#ClearHighlights:
line    2:
E119: Not enough arguments for function: prop_remove
Error detected while processing function go#lsp#ClearDiagnosticHighlights[1]..go#util#ClearHighlights:
line    2:
E119: Not enough arguments for function: prop_remove
```

Vim version
```
VIM - Vi IMproved 8.1 (2018 May 18, compiled Jun 15 2019 16:41:15)
Included patches: 1-875, 878, 884, 948, 1046, 1365-1368, 1382, 1401
```

Go version
```
go version go1.13.6 linux/amd64
```

I stripped down my `.vimrc` to this and tried again
```
" vim:foldmethod=marker
set nocompatible " be iMproved, required

set rtp+=~/.vim/bundle/Vundle.vim


if has('syntax') && has('eval')
  packadd! matchit
endif

" " Plugins {{{1
filetype off     " required for Vundle, re-enabled later
call vundle#begin()
Plugin 'VundleVim/Vundle.vim'
Plugin 'fatih/vim-go'

call vundle#end()            " required
filetype plugin indent on    " required
```

errors
```
Error detected while processing /home/skip/dotfiles/vim/.vim/bundle/vim-go/ftplugin/go.vim:
line   82:
E970: Unknown highlight group name: 'goDiagnosticError'
Error detected while processing function go#guru#ClearSameIds[1]..go#util#ClearHighlights:
line    2:
E119: Not enough arguments for function: prop_remove
Error detected while processing function go#lsp#ClearDiagnosticHighlights[1]..go#util#ClearHighlights
:
line    2:
E119: Not enough arguments for function: prop_remove
```

**Comment by sklukin** [comment]
The same problem on a fresh installation MasOs vim 8.2

**Comment by bhcleek** [comment]
@njkevlani thank you for reporting the output of `:scriptnames`. My output is similar: two go syntax files and ordered the same as yours with the Vim provided one after the one from my 1.22-rc.1 package.

Do you have a `goSameId` definition in `~/.vim/plugged/vim-go/syntax/go.vim` and in `/usr/share/vim/vim82/syntax/go.vim`?

@skipcloud the errors you saw about `prop_remove` should be resolved with #2662 

@skipcloud it's interesting that with your stripped down vimrc you saw a `Unknown highlight group name: 'goDiagnosticError'` error, but nothing about `goSameId` and that didn't see any such errors with your usual vimrc 🤔

**Comment by bhcleek** [comment]
I had a difficult time duplicating the issue about `Unknown highlight group` reported here.

Eventually, I discovered a replication path, though. The key was in @njkevlani's `:scriptnames` output. The order order of these files is different than it is on my machine:
```
39: /usr/share/vim/vim82/ftplugin/go.vim
40: ~/.vim/plugged/vim-go/indent/go.vim
41: /usr/share/vim/vim82/indent/go.vim
42: ~/.vim/plugged/vim-go/syntax/go.vim
43: /usr/share/vim/vim82/syntax/go.vim
```

A closer inspection of the vimrc revealed a difference from my usual vimrc: In the posted vimrc's, `syntax on` was called _before_ `filetype plugin on`. Once I changed my vimrc to do the same ordering, I was able to replicate the issue.

A fix is coming up. In the meantime, I am hopeful that anyone affected by this can workaround it by using `syntax on` or `syntax enable` prior to `filetype plugin on` in their vimrc.
